PsychSourceGL/Cohorts/FTGLTextRenderer/libptbdrawtext_ftgl.cpp

```diff
@@ -418,6 +422,10 @@ int PsychRebuildFont(fontCacheItem* fi)
         // Do not reset GL_MODELVIEW matrix after text drawing, so
         // we can read out the post-draw text cursor position from it:
         fi->faceM->setAdvance(true);
+
+	// set underlining, if requested in style
+	if (_fontStyle & 4)
```

Comment:
Most importantly though, this can be done more efficiently:

Atm. a change of fontstyle & 4 flag to en/disable underlining will trigger an expensive rebuilt of the font / creation of an extra copy of the font and all its glyph textures etc. although the only difference is if the underlines will be drawn inside OGLT::Texture.... stuff or not.

You could modify getForContext()'s use of _fontStyle to ignore bit 2 (== value 4) in _fontStyle, so a change of that bit will not trigger a font rebuild.

ie. in line 184, change
(fi->fontStyle == _fontStyle)
into
(fi->fontStyle == (_fontStyle & ~4))
so bit 2 gets ignored and then change line 230
fi->fontStyle = _fontStyle;
into
fi->fontStyle = _fontStyle & ~4;

Then replace the if (_fontStyle & 4) here by something like
fi->faceT->setDoUnderline(_fontStyle & 4);
and move this call into the proper place inside the proper place in PsychMeasureText() and PsychDrawText() - or an extra function to be called for both, so the setting gets toggled on/off as needed during measurement/drawing for a selected fi font object.

This should avoid unneccessary use of graphics/system memory and overhead for rebuilding fonts, given that there is a limited amount of these fontCacheItem slots per onscreen window. If somebody draws text in many different font/style combinations, this should help performance.

Reply:
And then verify that the underline is properly taken into account for bounding box calculations. Might be working already.

Reply:
Will do. Can't remember if i had a good reason for this after reading the code, or was just lazy :p